rust/kernel/lib.rs

```diff
@@ -140,5 +140,50 @@ fn panic(_info: &PanicInfo) -> ! {
     }
 }
 
+/// Calculates the offset of a field from the beginning of the struct it belongs to.
+#[macro_export]
+macro_rules! offset_of {
+    ($type:ty, $($f:tt)*) => {{
+        let tmp = core::mem::MaybeUninit::<$type>::uninit();
+        let ptr = tmp.as_ptr();
+        #[allow(unused_unsafe)]
```

Comment:
A comment on any `#[allow(...)]` would be nice, something like:

```suggestion
        // To avoid warnings when used in already-`unsafe` blocks.
        #[allow(unused_unsafe)]
```

Reply:
Done.